uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.js

```diff
@@ -156,11 +159,17 @@ define([
 
     // Init block
     (function init () {
-      $q.all([
-        loadStatuses(),
-        loadAbsenceTypes(),
-        loadAbsencePeriods()
-      ])
+      checkPermissions(sharedSettings.permissions.admin.administer)
+      .then(function (isAdmin) {
```

Comment:
Honestly I prefer using the `checkPermission` service , as it make the component independent of any outside properties/methods